src/test/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/helpers/MethodDescriptorHelperTest.kt

```diff
@@ -0,0 +1,34 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.helpers
+
+import com.intellij.psi.impl.PsiJavaParserFacadeImpl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+class MethodDescriptorHelperTest {
+
+    @ParameterizedTest
+    @MethodSource("primitiveFieldTypeTestGenerator")
+    fun primitiveFieldTypeTest(text: String, expected: String) {
+        val psiType = PsiJavaParserFacadeImpl.getPrimitiveType(text)
+        assertThat(generateFieldType(psiType)).isEqualTo(expected)
+    }
+
+    companion object {
+        @JvmStatic
+        private fun primitiveFieldTypeTestGenerator(): Stream<Arguments> {
```

Comment:
Is this the odd "Java descriptor" thing you've been talking about that hardcodes some types to letters?

Reply:
Yes. There is a context-free-grammar for it